frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,278 @@ export class OutputArtifactLoader {
     };
   }
 
+  public static async getMlmdContext(kfpPodName: string): Promise<Context> {
+    if (kfpPodName.split('-').length < 3) {
+      throw new Error('kfpPodName has fewer than 3 parts');
+    }
+
+    const pipelineName = kfpPodName
```

Comment:
It looks a bit unusual that you need to rely on Argo pod names.
Usually, the relevant information is present in explicit form. (E.g. pod labels and annotations)

Reply:
Agreed it'd be better to get pipelineName and runId explicitly, rather than trying to re-construct them. Do you know where they could be fetched?